turbopack/crates/turbo-tasks-backend/src/utils/ptr_eq_arc.rs

```diff
@@ -0,0 +1,47 @@
+use std::{
+    hash::{Hash, Hasher},
+    ops::Deref,
+    sync::Arc,
+};
+
+pub struct PtrEqArc<T>(Arc<T>);
+
+impl<T> PtrEqArc<T> {
+    pub fn new(value: T) -> Self {
+        Self(Arc::new(value))
+    }
+}
+
+impl<T> From<Arc<T>> for PtrEqArc<T> {
+    fn from(value: Arc<T>) -> Self {
+        Self(value)
+    }
+}
+
+impl<T> Deref for PtrEqArc<T> {
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
```

Comment:
IMO, this should deref to `Arc<T>`, which then derefs to `T`. In most cases with autoderef rules, the caller would see no difference, but it would make it possible to extract the `Arc<T>` if you need to for some reason.

The exception to that is if you're trying to hide the underlying `Arc<T>` for some reason, e.g. if you want to swap it for `triomphe::Arc` at some later point. But I don't think hiding this is worth it.